src/Lecture4.hs

```diff
@@ -232,9 +283,25 @@ our final result.
 If there's no some value (for example, there were not "Buy" products),
 you can return string "no value"
 -}
+showMaybeMax :: (Show a) => Maybe (Max a) -> String
+showMaybeMax Nothing = "no value"
+showMaybeMax (Just (Max x)) = show x
+
+showMaybeMin :: (Show a) => Maybe (Min a) -> String
+showMaybeMin Nothing = "no value"
+showMaybeMin (Just (Min x))= show x
 
 displayStats :: Stats -> String
-displayStats = error "TODO"
+displayStats (Stats sTotalPos sTotalSum sAbsMax sAbsMin sSellMax sSellMin sBuyMax sBuyMin sLong) = "\
+\Total positions:       : "++ (show . getSum $ sTotalPos) ++"\n\
+\Total final balance    : "++ (show . getSum $ sTotalSum) ++ "\n\
+\Biggest absolute cost  : "++ (show . getMax $ sAbsMax) ++ "\n\
+\Smallest absolute cost : "++ (show . getMin $ sAbsMin) ++ "\n\
+\Max earning            : "++ showMaybeMax sSellMax ++ "\n\
+\Min earning            : "++ showMaybeMin sSellMin ++ "\n\
+\Max spending           : "++ showMaybeMax sBuyMax ++ "\n\
+\Min spending           : "++ showMaybeMin sBuyMin ++ "\n\
+\Longest product name   : "++ unMaxLen sLong ++"\n"
```

Comment:
That's very cool usage of multiline strings 👏🏻 
Alternatively, you can use the `unlines` function and specify the list of strings. It will help to reduce some `\` noise and extra traling `\n` after each line.